common/src/main/scala/tsec/common/ByteUtils.scala

```diff
@@ -2,33 +2,75 @@ package tsec.common
 
 import java.security.MessageDigest
 
+import cats.Eq
+import cats.syntax.eq._
+
 object ByteUtils {
 
-  @inline def zeroByteArray(a: Array[Byte]): Unit =
+  /** Only exists because the scala stdlib decided
+    * iterators were the best way to implement a .find operation.
+    *
+    *  Will remove iff the new collections
+    *  remove that crap
+    *
+    */
+  @inline final def find[A: Eq](arr: Array[A], p: A => Boolean): Boolean =
```

Comment:
I think we don't need to constrain A here.

Reply:
True, will fix.